dashboard-lib/core-node-components/component.go

```diff
@@ -0,0 +1,175 @@
+package core_node_components
+
+import (
+	"github.com/K-Phoen/grabana/dashboard"
+	"github.com/K-Phoen/grabana/row"
+	"github.com/K-Phoen/grabana/stat"
+	"github.com/K-Phoen/grabana/table"
+	"github.com/K-Phoen/grabana/target/prometheus"
+	"github.com/K-Phoen/grabana/timeseries"
+	"github.com/K-Phoen/grabana/timeseries/axis"
+	"github.com/K-Phoen/grabana/variable/interval"
+	"github.com/K-Phoen/grabana/variable/query"
+)
+
+type Props struct {
+	PrometheusDataSource string
+	PlatformOpts         PlatformOpts
+}
+
+func vars(p Props) []dashboard.Option {
+	return []dashboard.Option{
+		dashboard.VariableAsInterval(
+			"interval",
+			interval.Values([]string{"30s", "1m", "5m", "15m", "30m", "1h", "6h", "12h"}),
+			interval.Default("15m"),
+		),
+		dashboard.VariableAsQuery(
+			"env",
+			query.DataSource(p.PrometheusDataSource),
+			query.IncludeAll(),
+			query.Request("label_values(up, env)"),
+			query.Sort(query.NumericalAsc),
+		),
+		dashboard.VariableAsQuery(
+			"cluster",
+			query.DataSource(p.PrometheusDataSource),
+			query.IncludeAll(),
+			query.Request(`label_values(up{env="$env"}, cluster)`),
+			query.Sort(query.NumericalAsc),
+		),
+		dashboard.VariableAsQuery(
+			"blockchain",
+			query.DataSource(p.PrometheusDataSource),
+			query.IncludeAll(),
+			query.Request(`label_values(up{env="$env", cluster="$cluster"}, blockchain)`),
+			query.Sort(query.NumericalAsc),
+		),
+		dashboard.VariableAsQuery(
+			"product",
+			query.DataSource(p.PrometheusDataSource),
+			query.IncludeAll(),
+			query.Request(`label_values(up{env="$env", cluster="$cluster", blockchain="$blockchain"}, product)`),
+			query.Sort(query.NumericalAsc),
+		),
+		dashboard.VariableAsQuery(
+			"network_type",
+			query.DataSource(p.PrometheusDataSource),
+			query.IncludeAll(),
+			query.Request(`label_values(up{env="$env", cluster="$cluster", blockchain="$blockchain", product="$product"}, network_type)`),
+			query.Sort(query.NumericalAsc),
+		),
+		dashboard.VariableAsQuery(
+			"component",
+			query.DataSource(p.PrometheusDataSource),
+			query.IncludeAll(),
+			query.Request(`label_values(up{env="$env", cluster="$cluster", blockchain="$blockchain", network_type="$network_type"}, component)`),
+			query.Sort(query.NumericalAsc),
+		),
+		dashboard.VariableAsQuery(
+			"service",
+			query.DataSource(p.PrometheusDataSource),
+			query.IncludeAll(),
+			query.Request(`label_values(up{env="$env", cluster="$cluster", blockchain="$blockchain", network_type="$network_type", component="$component"}, service)`),
+			query.Sort(query.NumericalAsc),
+		),
+		dashboard.VariableAsQuery(
+			"service_id",
+			query.DataSource(p.PrometheusDataSource),
+			query.Multiple(),
+			query.IncludeAll(),
+			query.Request(`label_values(health{env="$env", cluster="$cluster", blockchain="$blockchain", network_type="$network_type", component="$component", service="$service"}, service_id)`),
```

Comment:
This approach makes sense to me - the queries drill down by each level of granularity. Is there value in allowing some of them (i.e. product or service) to not necessarily need a higher level up if I want to quickly compare across envs? Or does grafana give us that for free?

Reply:
You don't have to specify all of them look at this example
<img width="1695" alt="image" src="https://github.com/smartcontractkit/chainlink/assets/16008249/7327ed88-c861-46f0-b105-df48e3a94b09">